test/coverlet.core.tests/Instrumentation/ModuleTrackerTemplateTests.cs

```diff
@@ -12,72 +9,62 @@ namespace Coverlet.Core.Tests.Instrumentation
         }
     }
 
-    [CollectionDefinition(nameof(ModuleTrackerTemplate))]
-    public class ModuleTrackerTemplateCollection : ICollectionFixture<ModuleTrackerTemplateTestsFixture>
-    {
-
-    }
-
-    [Collection(nameof(ModuleTrackerTemplate))]
-    public class ModuleTrackerTemplateTests : IDisposable
+    public class ModuleTrackerTemplateTests : IClassFixture<ModuleTrackerTemplateTestsFixture>, IDisposable
     {
-        private readonly MemoryMappedFile _mmap;
 
         public ModuleTrackerTemplateTests()
         {
-            ModuleTrackerTemplate.HitsArray = new int[4 + ModuleTrackerTemplate.HitsResultHeaderSize];
-            ModuleTrackerTemplate.HitsMemoryMapName = Guid.NewGuid().ToString();
-            ModuleTrackerTemplate.HitsFilePath = Path.Combine(Path.GetTempPath(), $"coverlet.test_{ModuleTrackerTemplate.HitsMemoryMapName}");
-
-            var size = (ModuleTrackerTemplate.HitsArray.Length + ModuleTrackerTemplate.HitsResultHeaderSize) * sizeof(int);
-
-            try
-            {
-                _mmap = MemoryMappedFile.CreateNew(ModuleTrackerTemplate.HitsMemoryMapName, size);
-            } 
-            catch (PlatformNotSupportedException)
-            {
-                _mmap = MemoryMappedFile.CreateFromFile(ModuleTrackerTemplate.HitsFilePath, FileMode.CreateNew, null, size);
-            }
+            File.Delete(ModuleTrackerTemplate.HitsFilePath);
         }
 
         public void Dispose()
         {
-            var hitsFilePath = ModuleTrackerTemplate.HitsFilePath;
-            _mmap.Dispose();
-            InstrumentationHelper.DeleteHitsFile(hitsFilePath);
+            File.Delete(ModuleTrackerTemplate.HitsFilePath);
         }
 
         [Fact]
         public void HitsFileCorrectlyWritten()
         {
-            RecordHits(1, 2, 0, 3);
+            ModuleTrackerTemplate.HitsArray = new[] { 1, 2, 0, 3 };
             ModuleTrackerTemplate.UnloadModule(null, null);
 
             var expectedHitsArray = new[] { 1, 2, 0, 3 };
-            Assert.Equal(expectedHitsArray, ReadHits());
+            Assert.Equal(expectedHitsArray, ReadHitsFile());
         }
 
         [Fact]
+        public void HitsFileWithDifferentNumberOfEntriesCausesExceptionOnUnload()
+        {
+            WriteHitsFile(new[] { 1, 2, 3 });
+            ModuleTrackerTemplate.HitsArray = new[] { 1 };
+            Assert.Throws<InvalidOperationException>(() => ModuleTrackerTemplate.UnloadModule(null, null));
+        }
+
+        [Fact(Skip="Failed CI Job: https://ci.appveyor.com/project/tonerdo/coverlet/builds/21145989/job/9gx5jnjs502vy1fv")]
         public void HitsOnMultipleThreadsCorrectlyCounted()
         {
-            for (int i = 0; i < 4; ++i)
+            ModuleTrackerTemplate.HitsArray = new[] { 0, 0, 0, 0 };
+            for (int i = 0; i < ModuleTrackerTemplate.HitsArray.Length; ++i)
             {
                 var t = new Thread(HitIndex);
                 t.Start(i);
-                t.Join();
```

Comment:
This line should be kept, because not waiting for the test threads to finish running makes the test unreliable. Then the Skip on the test can probably be removed.

Reply:
Maybe better track with issue and fix with a PR after merge.

Reply:
I generally prefer to do a full revert if possible, and then re-apply the part(s) that are still desired, but I wouldn't be opposed to either approach.

Reply:
Agreed, that is better. Optionally, the code this test is testing is no longer in the code base after changing to Interlocked.Increment in RecordHit so it could be removed altogether.